core/src/main/java/org/apache/iceberg/io/SupportsStorageCredentials.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.io;
+
+import java.util.List;
+
+public interface SupportsStorageCredentials {
```

Comment:
Could we add some JavaDocs on the interface? Don't think it's required on the actual interface methods since those are self explanatory, but something at the interface at level like "This interface is intended as an extension for FileIO implementations to be able to provide and retrieve storage credentials"

Reply:
yeah good point, I've added that